---
title: "GRBC_DeSeq2"
author: "Dylan Baker"
date: "6/2/2020"
output:
  html_document
---

```{r}
require("knitr")
## Loading required package: knitr
#same wd for all chunks
#opts_knit$set(root.dir = "~/Desktop/CRUDE_EEB401")

knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = FALSE,
                      include = TRUE,
                      collapse = FALSE,
                      message=FALSE,
                      warning=FALSE,
                      dependson = NULL,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      #Figure options
                      dev = c("png", "pdf"),
                      fig.path="outfiles_demo/",  
                      fig.align = "center", 
                      fig.width = 8,
                      fig.height = 6
)

```
```{r setup}

#libraries and packages
list.of.packages <- c(
  "knitr",
  "ggplot2",
  "vegan",
  "dplyr",
  "DESeq2",
  "scales",
  "grid",
  "reshape2",
  "plyr",
  "phyloseq",
  "magrittr",
  "geosphere",
  "matrixStats",
  "data.table",
  "DT",
  "tidyverse",
  "ashr",
  "pheatmap"
  ) 
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])] 

if(length(new.packages)) install.packages(new.packages)
if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
BiocManager::install("Phyloseq")
BiocManager::install("DESeq2")
BiocManager::install("apeglm")
BiocManager::valid("DESeq2")

#source codes
#source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R", local = TRUE)
#source("https://bioconductor.org/biocLite.R")
#biocLite("DESeq2")
#biocLnite("annotate")
#biocLite('phyloseq')

library(ggplot2) 
library(vegan) 
library(dplyr) 
library(scales) 
library(grid) 
library(reshape2)
library(plyr)
library(phyloseq) 
library(magrittr)
library(geosphere)
library(matrixStats)
library(data.table)
library(DT)
library(DESeq2)
library(colorspace)
library(tidyverse)
library(apeglm)
library(ashr)
library(pheatmap)
library(RColorBrewer)
#shortened theme
#theme_set(theme_bw())


#set seed so all random number generation is the same
set.seed(05042019)

```
### Importing Data and Creating Phyloseq Object
Import data (three files: shared (OTUfile), taxonomy (TAXfile), and metadata (mapfile)) Then combining data files into one phyloseq object for analysis (all_data)

# Research Goals

**Research Question** Look for differences between the associated microbes of host algae after 3 and 31 days of exposure to water from three George Reserve Ponds. I want to compare the difference between the natural community at day 0 and specific associated bacteria with the 5 algae at day 3 and day 31. How does the community change over time? Is there a specific genus of bacteria that is selected for after 31 days? Are there genera present in the natural pond communities that are not present in host phycosphere samples?

*Hypothesis 1* I hypothesize that there will be genera present at a higher relative abundance in host phycospheres that will not be present in the natural community

*Hypothesis 2* I hypothesize that there will be specific genera that are recruited by algae and that these genera are responsible for carbon cycling, vitamin production, algicide production, or other beneficial activities that will increase the fitness of the host algae.

**This Analysis**

**Results**

```{r data_import_create_phylo, include=FALSE}
#Create a list of jar numbers and their respective host species.
jar_num <- as_tibble(paste0("J", 1:45, "-")) %>%
  rename(value = "jar_no") %>%
  mutate(host_species = ifelse(str_detect(jar_no, paste0("J", 1:9, "-")), "Chlorella",
                        ifelse(str_detect(jar_no, paste0("J", 10:18, "-")), "Coelastrum",
                        ifelse(str_detect(jar_no, paste0("J", 19:27, "-")), "Scenedesmus",
                        ifelse(str_detect(jar_no, paste0("J", 28:36, "-")), "Monoraphidium",
                        ifelse(str_detect(jar_no, paste0("J", 37:45, "-")), "Selenastrum", "NA"))))))
jar_no_grbc <- read_csv(file = "Jar_Number_Correction.csv", na = "empty") %>%
  mutate(jar = paste0(jar_no, "-")) %>%
  select(Isolate_Number, jar)

# Assign variables for imported data
sharedfile = "Phycosphere_stability.file.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.shared.txt"
taxfile = "Phycosphere_stability.file.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.cons.taxonomy.txt"

# Import mothur data and write a sample name file
mothur_data <- import_mothur(mothur_shared_file = sharedfile,
  mothur_constaxonomy_file = taxfile)
sample_names(mothur_data)


write.table(sample_names(mothur_data), file = "Phyco.sample_names.tsv", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)

# Import sample metadata
map <- read.table("Phyco.sample_names.tsv", header=FALSE, sep="\t") %>%
  rename( V1 = "Group") %>%
  #Add Host Species Information
    mutate(host_species = ifelse(str_detect(Group, paste(jar_num$jar_no[1:9], collapse = "|")), "Chlorella",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[10:18], collapse = "|")), "Coelastrum",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[19:27], collapse = "|")), "Scenedesmus",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[28:36], collapse = "|")), "Monoraphidium",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[37:45], collapse = "|")), "Selenastrum",
                        ifelse(str_detect(Group, "Blank"),
"control",
                        ifelse(str_detect(Group, "P"),
"natural","NA")))))))) %>%
  #Add day info e.g. Day 3 or Day 31
  mutate(time = ifelse(str_detect(Group, "D0-"), "0",
                       ifelse(str_detect(Group, "D3-"), "3",
                       ifelse(str_detect(Group, "D10-"), "10",
                       ifelse(str_detect(Group, "D17-"), "17",
                       ifelse(str_detect(Group, "D24-"), "24",
                       ifelse(str_detect(Group, "D31-"), "31", "NA")))
                                            )))) %>%
  #Add replicate info (R doesn't support index skipping like python, so this will have to do)
  mutate(replicate = ifelse(
  str_detect(Group,paste(jar_num$jar_no[c(1,4,7,10,13,16,19,22,25,28,31,34,37,40,43)], collapse = "|")), "1",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(2,5,8,11,14,17,20,23,26,29,32,35,38,41,44)], collapse = "|")), "2",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(3,6,9,12,15,18,21,24,27,30,33,36,39,42,45)], collapse = "|")), "3",
  ifelse(str_detect(Group, "R1") | str_detect(Group, "-1"), "1", 
  ifelse(str_detect(Group, "R2") | str_detect(Group, "-2"), "2",
  ifelse(str_detect(Group, "R3"), "3", "NA"))))))) %>%
  #Add Pond info
  mutate(pond = ifelse(
  str_detect(Group,paste(jar_num$jar_no[c(1:3,10:12,19:21,28:30,37:39)], collapse = "|")), "1",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(4:6,13:15,22:24,31:33,40:42)], collapse = "|")), "2",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(7:9,16:18,25:27,34:36,43:45)], collapse = "|")), "3", 
    ifelse(str_detect(Group, "P1"), "1", 
    ifelse(str_detect(Group, "P2"), "2",
    ifelse(str_detect(Group, "P3"), "3", "NA"))))))) %>%
  #Add filter type info 
  mutate(filter = ifelse(str_detect(Group, "022um"), "022um",
         ifelse(str_detect(Group, "30um") | str_detect(Group, "03um"), "3um", "NA")))


map <- sample_data(map)
#head(map)
rownames(map) <- map$Group

phylo_all <- merge_phyloseq(mothur_data, map) # the final phyloseq object to work with
phylo_all
```
## Filter by Group
```{r group_filter}


# change Rank 1-7 to actual taxonomy labels
colnames(tax_table(phylo_all)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")

# check that it worked
rank_names(phylo_all)
```

```{r seq_counts}
# make a data frame with a column for the sequence counts of each sample
phylo_sum_df <- data.frame(sum = sample_sums(phylo_all))
datatable(phylo_sum_df, 
          options = list(pageLength = 10),
          caption = "Sum of sequences per sample"
          )
# From sorting this table by decreasing sequence count, we can see that there were a few samples that had only a small number of reads. We will remove these (prune) from our data for analysis later on.

```
Next we will calculate summary statistics describing the sequence counts with the summary() function. We will then represent these stats with a boxplot and a histogram
```{r summary}

#summary statistics of sequence counts
summary(phylo_sum_df$sum)

#histogram of sample read counts
ggplot(phylo_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "purple", binwidth = 500) +
  ggtitle("Distribution of sample sequencing depth") + 
  xlab("Read counts") +
  geom_vline(xintercept = 1000, linetype="dotted", color = "blue", size=0.5) +
  geom_vline(xintercept = 500, linetype="dotted", color = "red", size=0.5) +
  theme(axis.title.y = element_blank())

```

Here we will filtering out (prune) samples with low read counts (<1000). This is called the sample sequencing depth.  We will do this by creating a new pruned phyloseq object (`pruned_all_data`). 

```{r pruning_low_read_samples}

# creating a phyloseq object called pruned_all_data that only includes samples with a sequence depth >1000
pruned_phyco <- prune_samples(sample_sums(phylo_all) > 1000, phylo_all)
#pruned_all_data


#creating a dataframe with a column for the sequence counts of each sample
pruned_phyco_sum_df <- data.frame(sum = sample_sums(pruned_phyco))
#histogram of sample read counts
ggplot(pruned_phyco_sum_df, aes(x = sum)) + 
  geom_histogram(color = "black", fill = "purple", binwidth = 500) +
  ggtitle("Distribution of pruned sample sequencing depth (>500 reads)") + 
  xlab("Read counts") +
  geom_vline(xintercept = 1000, linetype="dotted", color = "blue", size=0.5) +
  theme(axis.title.y = element_blank())
```

## Removing non-bacterial reads

We will clean up our dataset to only include OTUs that are Bacterial, removing both mitochondiral and chloroplast OTUs. Here, we also remove test, mock community, and negative water controls that were sequenced. At the end of the chunk we create two new phyloseq objects. The first is a copy of our current one that we now call avbot.raw. Then in the second one, we log-transform the sample counts and write that into the phyloseq object called `avbot`. Why might we want to log-transform the sample counts?

```{r Subset_Phyco}

#plot_bar(phyco, x="sample_type", fill="Phylum") # takes a long time

pruned_phyco_clean <- pruned_phyco %>%
  subset_taxa(
    Domain == "Bacteria" &
    Family  != "mitochondria" &
    Family   != "Chloroplast"
  )%>%
  subset_samples(
    time == "3" | time == "31" | time == "0" & filter == "022um"
  )

pruned_phyco_clean

pruned_phyco_clean.raw <- pruned_phyco_clean
scaled_phyco_clean <- transform_sample_counts(pruned_phyco_clean.raw, function(x) log(1 + x))
sampsums <- data.frame(sums = sample_sums(scaled_phyco_clean))

ggplot(sampsums, aes(x = sums)) + geom_histogram(binwidth = 5)

#Betadiv pcoa Bray-Curtis all data
physeq.pcoa <- 
  ordinate(
    physeq = scaled_phyco_clean,
    method = "PCoA",
    distance = "bray"
  )

physeq.pcoa.vectors <- data.frame(physeq.pcoa$vectors[, 1:4])
#physeq.pcoa.vectors$Duplicates <- row.names(physeq.pcoa.vectors)
physeq.pcoa.df <- merge(physeq.pcoa.vectors,data.frame(sample_data(scaled_phyco_clean)),by="row.names")
bray_values <- physeq.pcoa$values
bray_rel_eigens <- bray_values$Relative_eig
bray_rel_eigen1 <- bray_rel_eigens[1]
bray_rel_eigen1_percent <- round(bray_rel_eigen1 * 100, digits = 1)
bray_rel_eigen2 <- bray_rel_eigens[2]
bray_rel_eigen2_percent <- round(bray_rel_eigen2 * 100, digits = 1)
bray_rel_eigen3 <- bray_rel_eigens[3]
bray_rel_eigen3_percent <- round(bray_rel_eigen3 * 100, digits = 1)
bray_rel_eigen4 <- bray_rel_eigens[4]
bray_rel_eigen4_percent <- round(bray_rel_eigen4 * 100, digits = 1)
bray_axis1 <- paste("PCoA1:",bray_rel_eigen1_percent,"%")
bray_axis2 <- paste("PCoA2:",bray_rel_eigen2_percent,"%")
bray_axis3 <- paste("PCoA3:",bray_rel_eigen3_percent,"%")
bray_axis4 <- paste("PCoA4:",bray_rel_eigen4_percent,"%")
PCoA_title <- paste("Bray-Curtis,",ntaxa(scaled_phyco_clean),"OTUs")

physeq.pcoa.df %>% 
  ggplot(aes(Axis.1, Axis.2, color = host_species, shape = time)) +
  xlab(bray_axis1) + ylab(bray_axis2) + ggtitle(PCoA_title) +
  geom_point(alpha=0.9) + theme_bw() + #ylim(1, -1) + xlim(1, -1) 
  scale_shape_manual(name = "time", breaks=c("0","3", "31"),
                     labels = c("Day 0","Day 3", "Day 31"), 
                     values = c("0" = 18, "3" = 15, "31" = 16 )) +
  scale_color_manual(name = "host_species", breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "control", "natural"),
                     labels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Control", "Natural"), 
                     values = c("Chlorella" = "#7FC97F", "Coelastrum" = "#BEAED4", "Scenedesmus" ="#E0115F", "Monoraphidium" = "#FDC086", "Selenastrum" = "#386CB0", "control" = "#35978f", "natural" = "#bf812d")) +
  theme(plot.title = element_text(face="bold", size=8),  #Set the plot title
        axis.text.x = element_text(colour="black", vjust=0.5, size=8), 
        axis.text.y = element_text(colour="black", vjust=0.5, size=8),
        axis.title.x = element_text(face="bold", size=8),
        axis.title.y = element_text(face="bold", size=8),
        legend.title = element_text(size=12, face="bold"),
        legend.text = element_text(size = 12),
        ###LEGEND TOP RIGHT CORNER
        legend.position = "right") 

```
```{r phyla colors}
phyla.colors <- c(Acidobacteria = "grey26", 
                  Actinobacteria = "palevioletred2", 
                  Alphaproteobacteria = "steelblue4", 
                  Armatimonadetes = "red", 
                  Bacteroidetes = "darkorange", 
                  "BD1-5" = "chartreuse", 
                  Betaproteobacteria = "royalblue", 
                  Caldiserica = "black",
                  "WS1" = "red", 
                  "WPS-2" = "aquamarine1", 
                  "Candidate_division_OD1" = "#6DDE88", 
                  "Candidate_division_OP3" = "yellow1", 
                  "Candidate_division_OP8" = "goldenrod1", 
                  "Candidate_division_OP11" = "chocolate4", 
                  "Candidate_division_SR1" = "tan3", 
                  "Candidate_division_TM7" = "skyblue1", 
                  "Candidate_division_WS3" = "magenta", 
                  "Candidate_Division_CK_1C4_19" = "darkseagreen", 
                  "Candidate_Division_NPL_UPA2" = "darkmagenta", 
                  "Candidate_Division_RF3" = "bisque", 
                  "Candidate_Division_TA06" = "cadetblue",  
                  "Candidate_Division_TM6" = "plum", 
                  Chlamydiae="violet", 
                  Chlorobi="cyan2", 
                  Chloroflexi="darkgreen", 
                  Crenarchaeota = "khaki1", 
                  phycobacteria = "chartreuse3", 
                  Deferribacteres = "slateblue3", 
                  "Deinococcus-Thermus" = "violetred", 
                  Dictyoglomi = "cornsilk4", 
                  Deltaproteobacteria = "deepskyblue", 
                  "Deinococcus-Thermus" = "lightskyblue1", 
                  Elusimicrobia = "violetred4",
                  Epsilonproteobacteria = "lightskyblue", 
                  Euryarchaeota = "khaki2", 
                  Fibrobacteres = "hotpink", 
                  Firmicutes = "blue4", 
                  FGL7S = "palevioletred1", 
                  Fusobacteria = "slateblue1",
                  Euglenozoa = "#652926", 
                  Gammaproteobacteria = "plum2", 
                  Gemmatimonadetes="black", 
                  GOUTA4 = "plum1", 
                  "Hyd24-12" = "sienna2",
                  Ignavibacteriae = "sienna2", 
                  JTB23 = "seashell2", 
                  Lentisphaerae = "cyan4", 
                  "MVP_21" = "lightskyblue1", 
                  Nitrospirae = "yellow1", 
                  "NPL-UPA2"="#652926", 
                  Hydrogenedentes = "mediumpurple4", 
                  Planctomycetes = "mediumorchid3", 
                  Proteobacteria = "deepskyblue", 
                  Retaria ="lightsalmon3", 
                  Kiritimatiellaeota = "lightskyblue2", 
                  Patescibacteria = "orangered", 
                  Spirochaetes = "gold2", 
                  Spirochaetae = "gold3", 
                  Synergistetes = "olivedrab1", 
                  Tenericutes="pink", 
                  Thaumarchaeota = "khaki3", 
                  BRC1 = "chocolate1", 
                  Halanaerobiaeota = "lightsalmon3",
                  Dependentiae = "rosybrown3", 
                  TM6 = "olivedrab", 
                  unclassified = "grey", 
                  Verrucomicrobia = "purple4", 
                  Epsilonbacteraeota = "palegreen", 
                  Bacteria_unclassified = "darkgrey")
get_random_grid_colors <- function(ncolor,seed = sample_seq(1, 1000000, 1)) {
  require(uniformly)
  set.seed(seed)
  print(seed)
  ngrid <- ceiling(ncolor^(1/3))
  x <- seq(0,1,length=ngrid+1)[1:ngrid]
  dx <- (x[2] - x[1])/2
  x <- x + dx
  origins <- expand.grid(x,x,x)
  nbox <- nrow(origins) 
  RGB <- vector("numeric",nbox)
  for(i in seq_len(nbox)) {
    rgb <- runif_in_cube(n=1,d=3,O=as.numeric(origins[i,]),r=dx)
    RGB[i] <- rgb(rgb[1,1],rgb[1,2],rgb[1,3])
  }
  index <- sample(seq(1,nbox),ncolor)
  RGB[index]
} 

colors <- get_random_grid_colors(15, seed = 904057)
```

```{r DeSeq2}
phyco_deseq <- phyloseq_to_deseq2(pruned_phyco_clean.raw, ~ Group)
phyco_deseq$Group <- factor(paste0(phyco_deseq$host_species, phyco_deseq$time))
phyco_deseq <- DESeq(phyco_deseq, test = "Wald", fitType = "parametric")

#A simple plot of the concentration of counts in a single sample over the sum of counts per gene. Not technically the same as "sparsity", but this plot is useful diagnostic for datasets which might not fit a negative binomial assumption: genes with many zeros and individual very large counts are difficult to model with the negative binomial distribution.
plotSparsity(phyco_deseq)

res = results(phyco_deseq)
alpha = 0.01
sigtab = res[which(res$padj < alpha),]
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(pruned_phyco_clean.raw)[rownames(sigtab), ], "matrix"))
head(sigtab)
plotMA(res, ylim = c(-2,2))

theme_set(theme_bw())
scale_fill_discrete <- function(palname = "Set1", ...) {
    scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab$log2FoldChange, sigtab$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab$Phylum = factor(as.character(sigtab$Phylum), levels=names(x))

# Genus order
x = tapply(sigtab$log2FoldChange, sigtab$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab$Genus = factor(as.character(sigtab$Genus), levels=names(x))

results(phyco_deseq)
results(phyco_deseq, contrast = c("Group", "Chlorella3", "Chlorella31"))


#Shrinkage of effect size (LFC estimates) is useful for visualization and ranking of genes. To shrink the LFC, we pass the dds object to the function lfcShrink. Below we specify to use the apeglm method for effect size shrinkage (Zhu, Ibrahim, and Love 2018), which improves on the previous estimator.

# because we are interested in 5 different host species with time, we set 'coef=10'
resAsh <- lfcShrink(phyco_deseq, coef = "Group_Chlorella31_vs_Chlorella3", type="ashr") 
resLFC <- lfcShrink(phyco_deseq, coef = "Group_Chlorella31_vs_Chlorella3", type = "apeglm")
resNorm <- lfcShrink(phyco_deseq, coef = 10, type = "normal")

par(mfrow=c(1,4), mar=c(4,4,2,1))
xlim <- c(1,1e5); ylim <- c(-3,3)
plotMA(resLFC, xlim=xlim, ylim=ylim, main="apeglm")
plotMA(resAsh, xlim=xlim, ylim=ylim, main="ashr")
plotMA(resNorm, xlim=xlim, ylim=ylim, main = "norm" )
plotMA(res, xlim=xlim, ylim=ylim, main = "unnormalized" )




sigtabLFC = resLFC[which(res$padj < alpha),]
sigtabLFC = cbind(as(sigtab, "data.frame"), as(tax_table(pruned_phyco_clean.raw)[rownames(sigtabLFC), ], "matrix"))
head(sigtabLFC)
# Phylum order
x = tapply(sigtabLFC$log2FoldChange, sigtabLFC$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtabLFC$Phylum = factor(as.character(sigtabLFC$Phylum), levels=names(x))
# Genus order
x = tapply(sigtabLFC$log2FoldChange, sigtabLFC$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtabLFC$Genus = factor(as.character(sigtabLFC$Genus), levels=names(x))




ggplot(sigtabLFC, aes(x=Genus, y=log2FoldChange, color=Phylum)) + 
  theme_bw() +
  geom_point(alpha=0.9) +
  scale_size_continuous("OTU counts") +
  coord_flip() +
  scale_color_manual(values = colors) +
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5), legend.position = "top") +
  ylab("<-- Coelastrum   log2FoldChange   Chlorella -->")+
  theme(axis.text.x=element_text(angle=90, vjust=0.1, face="bold.italic"))+
  ggtitle("Over/Under-Represented OTUs Coelastrum vs Chlorella\nLog2FoldChange p<0.05 (DeSeq2)") + theme(plot.title = element_text(lineheight=1))


sigtabAsh = resAsh[which(res$padj < alpha),]
sigtabAsh = cbind(as(sigtab, "data.frame"), as(tax_table(pruned_phyco_clean.raw)[rownames(sigtabAsh), ], "matrix"))

head(sigtabAsh)
# Phylum order
x = tapply(sigtabAsh$log2FoldChange, sigtabAsh$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtabAsh$Phylum = factor(as.character(sigtabAsh$Phylum), levels=names(x))
# Genus order
x = tapply(sigtabAsh$log2FoldChange, sigtabAsh$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtabAsh$Genus = factor(as.character(sigtabAsh$Genus), levels=names(x))
```

```{r phreatmap}
# this gives log2(n + 1)
ntd <- normTransform(phyco_deseq)
select <- order(rowMeans(counts(phyco_deseq, normalized=TRUE)),
                decreasing=TRUE) [1:20]
df <- as.data.frame(colData(phyco_deseq)[,c("time","host_species")])

pheatmap(assay(ntd)[select,], cluster_rows=FALSE, show_rownames=TRUE,
         cluster_cols=TRUE, annotation_col=df)

vsd <- vst(phyco_deseq, blind = FALSE)



```