---
title: "GRBC_Abundance_Recolonization"
author: "Dylan Baker"
date: "2/26/2020"
output:
  html_document:
    code_folding: hide
    highlight: haddock
    keep_md: yes
    number_sections: yes
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
      toc_depth: 3
  pdf_document:
    toc: yes
---
```{r}
require("knitr")
## Loading required package: knitr

knitr::opts_chunk$set(eval = TRUE, 
                      echo = TRUE, 
                      cache = FALSE,
                      include = TRUE,
                      collapse = FALSE,
                      message=FALSE,
                      warning=FALSE,
                      dependson = NULL,
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      #Figure options
                      dev = c("png", "pdf"),
                      fig.path="./figures",  
                      fig.align = "center", 
                      fig.width = 8,
                      fig.height = 6
)

```
``` {r setup}
list.of.packages <- c(
  "knitr",
  "dplyr",
  "data.table",
  "readr",
  "tidyverse",
  "stringr",
  "purrr",
  "broom",
  "uniformly",
  "gtools",
  "gridExtra",
  "ggpubr",
  "igraph",
  "phyloseq",
  "tidytext",
  "RColorBrewer",
  "reprex",
  "forcats",
  "patchwork",
  "xlsx"
)
new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])] 
if(length(new.packages)) install.packages(new.packages)

if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")
#BiocManager::install("phyloseq")


library(dplyr) 
library(reshape2)
library(data.table)
library(readr)
library(tidyverse)
library(stringr)
library(purrr)
library(broom)
library(phyloseq)
library(uniformly)
library(gtools)
library(gridExtra)
library(ggpubr)
library(igraph)
library(tidytext)
library(RColorBrewer)
library(reprex)
library(forcats)
library(patchwork)
library(xlsx)
#set seed so all random number generation is the same
set.seed(05042019)

#Set a good color palette
#https://stackoverflow.com/questions/15282580/how-to-generate-a-number-of-most-distinctive-colors-in-r?answertab=active#tab-top
get_random_distinct_colors <- function(ncolor,seed = sample_seq(1, 1000000, 1)) {
  require(uniformly)
  set.seed(seed)
  rgb_mat <- runif_in_cube(n=ncolor,d=3,O=rep(0.5,3),r=0.5)
  rgb(r=rgb_mat[,1],g=rgb_mat[,2],b=rgb_mat[,3])
}


#https://martin.ankerl.com/2009/12/09/how-to-create-random-colors-programmatically/

get_distinct_hues <- function(ncolor,s=0.5,v=0.95,seed=40) {
  golden_ratio_conjugate <- 0.618033988749895
  set.seed(seed)
  h <- runif(1)
  H <- vector("numeric",ncolor)
  for(i in seq_len(ncolor)) {
    h <- (h + golden_ratio_conjugate) %% 1
    H[i] <- h
  }
  hsv(H,s=s,v=v)
}

get_random_grid_colors <- function(ncolor,seed = sample_seq(1, 1000000, 1)) {
  require(uniformly)
  set.seed(seed)
  print(seed)
  ngrid <- ceiling(ncolor^(1/3))
  x <- seq(0,1,length=ngrid+1)[1:ngrid]
  dx <- (x[2] - x[1])/2
  x <- x + dx
  origins <- expand.grid(x,x,x)
  nbox <- nrow(origins) 
  RGB <- vector("numeric",nbox)
  for(i in seq_len(nbox)) {
    rgb <- runif_in_cube(n=1,d=3,O=as.numeric(origins[i,]),r=dx)
    RGB[i] <- rgb(rgb[1,1],rgb[1,2],rgb[1,3])
  }
  index <- sample(seq(1,nbox),ncolor)
  RGB[index]
} 

#Get hexadecimal codes for Accent brewer palette
display.brewer.pal(n = 5, 'Accent')
brewer.pal(n = 5, 'Accent') # "#7FC97F" "#BEAED4" "#FDC086" "#FFFF99" "#386CB0"
host_colors <- c("#7FC97F", "#BEAED4","#E0115F", "#FDC086","#386CB0")

```

``` {r Import}

#Create a list of jar numbers and their respective host species.
jar_num <- as_tibble(paste0("J", 1:45, "-")) %>%
  rename("jar_no" = value) %>%
  mutate(host_species = ifelse(str_detect(jar_no, paste0("J", 1:9, "-")), "Chlorella",
                        ifelse(str_detect(jar_no, paste0("J", 10:18, "-")), "Coelastrum",
                        ifelse(str_detect(jar_no, paste0("J", 19:27, "-")), "Scenedesmus",
                        ifelse(str_detect(jar_no, paste0("J", 28:36, "-")), "Monoraphidium",
                        ifelse(str_detect(jar_no, paste0("J", 37:45, "-")), "Selenastrum", ""))))))
jar_no_grbc <- read_csv(file = "Jar_Number_Correction.csv", na = "empty") %>%
  mutate(jar = paste0(jar_no, "-")) %>%
  select(Isolate_Number, jar)

#Remove confidence scores at the end of each taxonomic ran
#Import the data
GRBC_Silva_132_tax <- read_tsv(file="GRBC_renamed.nr_v132.wang.taxonomy") %>%
		rename_all(tolower) %>%  mutate(taxonomy=str_replace_all(string=taxonomy, pattern="\\(\\d*\\)", replacement="")) %>%
		mutate(taxonomy=str_replace_all(string=taxonomy , pattern=";$", replacement="")) %>%
		separate(taxonomy, into=c("domain", "phylum", "class", "order", "family", "genus"), sep=";") %>%
  inner_join(jar_no_grbc, by = c("otu" = "Isolate_Number")) %>%
  #Filter out genera that were not in the database, not helpful for our analysis
  filter(!str_detect(genus, "unclassified")) %>%
  filter(!str_detect(genus, "uncultured")) %>%
  select("otu", "jar", everything()) %>%
  #Replace DF with D31
  lapply(function(x) {
    gsub("DF", "D31", x)
  }) %>%
  as.tibble(.) %>%
  mutate(host_species = ifelse(str_detect(jar, paste(jar_num$jar_no[1:9], collapse = "|")), "Chlorella",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[10:18], collapse = "|")), "Coelastrum",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[19:27], collapse = "|")), "Scenedesmus",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[28:36], collapse = "|")), "Monoraphidium",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[37:45], collapse = "|")), "Selenastrum", "NA"))))))

GRBC_Freshwater2016_tax <- read_tsv(file="GRBC_renamed.FreshTrain18Aug2016.wang.taxonomy") %>%
		rename_all(tolower) %>%  mutate(taxonomy=str_replace_all(string=taxonomy, pattern="\\(\\d*\\)", replacement="")) %>%
		mutate(taxonomy=str_replace_all(string=taxonomy , pattern=";$", replacement="")) %>%
		separate(taxonomy, into=c("domain", "phylum", "class", "order", "family", "genus"), sep=";") %>%
  inner_join(jar_no_grbc, by = c("otu" = "Isolate_Number")) %>%
  #Filter out genera that were not in the database, not helpful for our analysis
  filter(!str_detect(genus, "unclassified")) %>%
  filter(!str_detect(genus, "uncultured")) %>%
  select("otu", "jar", everything()) %>%
  #Replace DF with D31
  lapply(function(x) {
    gsub("DF", "D31", x)
  }) %>%
  as.tibble(.) %>%
  mutate(host_species = ifelse(str_detect(jar, paste(jar_num$jar_no[1:9], collapse = "|")), "Chlorella",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[10:18], collapse = "|")), "Coelastrum",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[19:27], collapse = "|")), "Scenedesmus",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[28:36], collapse = "|")), "Monoraphidium",
                        ifelse(str_detect(jar, paste(jar_num$jar_no[37:45], collapse = "|")), "Selenastrum", "NA"))))))

#Combine silva and freshwater dataframes
GRBC_all_tax <- merge(GRBC_Freshwater2016_tax, GRBC_Silva_132_tax, all = TRUE) %>%
  #Add a column to distinguish D3 and D31
  mutate(time = ifelse(str_detect(otu, "_D31"), "D31", "D3"),
         color = 
          ifelse(host_species == "Chlorella", "#7FC97F", 
          ifelse(host_species == "Coelastrum", "#BEAED4",
          ifelse(host_species == "Scenedesmus", "#E0115F",
          ifelse(host_species == "Monoraphidium", "#FDC086",
          ifelse(host_species == "Selenastrum", "#386CB0", "black")))))) %>%
  #Add replicate info (R doesn't support index skipping like python, so this will have to do)
  mutate(replicate = ifelse(
  str_detect(jar,paste(jar_num$jar_no[c(1,4,7,10,13,16,19,22,25,28,31,34,37,40,43)], collapse = "|")), "1",
                     ifelse(
  str_detect(jar, paste(jar_num$jar_no[c(2,5,8,11,14,17,20,23,26,29,32,35,38,41,44)], collapse = "|")), "2",
                     ifelse(
  str_detect(jar, paste(jar_num$jar_no[c(3,6,9,12,15,18,21,24,27,30,33,36,39,42,45)], collapse = "|")), "3", "NA")))) %>%
  #Add Pond info
  mutate(pond = ifelse(
  str_detect(jar, paste(jar_num$jar_no[c(1:3,10:12,19:21,28:30,37:39)], collapse = "|")), "1",
                     ifelse(
  str_detect(jar, paste(jar_num$jar_no[c(4:6,13:15,22:24,31:33,40:42)], collapse = "|")), "2",
                     ifelse(
  str_detect(jar, paste(jar_num$jar_no[c(7:9,16:18,25:27,34:36,43:45)], collapse = "|")), "3", "NA")))) %>%
  #genus name is too long, ruins later graphs.
  mutate(genus=str_replace_all(string=genus,
    pattern="Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", replacement="A-N-P-R")) %>%
  #Rearrage columns in a readable order
  select(otu, jar, replicate, time, pond, color, everything())

Phycosphere_tax <- read_tsv(file="Phycosphere_stability.file.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.0.03.cons.taxonomy.txt") %>%
		rename_all(tolower) %>%  mutate(taxonomy=str_replace_all(string=taxonomy, pattern="\\(\\d*\\)", replacement="")) %>%
		mutate(taxonomy=str_replace_all(string=taxonomy , pattern=";$", replacement="")) %>%
		separate(taxonomy, into=c("domain", "phylum", "class", "order", "family", "genus"), sep=";") %>%
  filter(!str_detect(genus, "unclassified")) %>%
  filter(!str_detect(genus, "uncultured")) %>%
  filter(!str_detect(genus, "Unknown")) 
	
Phycosphere_otu <- read_tsv("Phycosphere_stability.file.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.an.unique_list.shared.txt",col_types=cols(Group=col_character())) %>%
	select(-label, -numOtus) %>%
 	pivot_longer(cols=-Group, names_to="otu", values_to="count") 

#We need the number from this output to standardize counts below.
Phycosphere_counts <- Phycosphere_otu %>%
  #make the data table recognize samples
  group_by(Group) %>%
  summarize(n = sum(count)) %>%
  ungroup() %>%
  rename("total_count_group" = n)

Phycosphere_otu_count <- inner_join(Phycosphere_otu, Phycosphere_counts, by = "Group") %>%
  mutate(rel_abund = count/total_count_group) %>%
  #Add Host Species Information
    mutate(host_species = ifelse(str_detect(Group, paste(jar_num$jar_no[1:9], collapse = "|")), "Chlorella",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[10:18], collapse = "|")), "Coelastrum",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[19:27], collapse = "|")), "Scenedesmus",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[28:36], collapse = "|")), "Monoraphidium",
                        ifelse(str_detect(Group, paste(jar_num$jar_no[37:45], collapse = "|")), "Selenastrum", "Natural Community")))))) %>%
  #Add replicate info (R doesn't support index skipping like python, so this will have to do)
  mutate(replicate = ifelse(
  str_detect(Group,paste(jar_num$jar_no[c(1,4,7,10,13,16,19,22,25,28,31,34,37,40,43)], collapse = "|")), "1",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(2,5,8,11,14,17,20,23,26,29,32,35,38,41,44)], collapse = "|")), "2",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(3,6,9,12,15,18,21,24,27,30,33,36,39,42,45)], collapse = "|")), "3",
  ifelse(str_detect(Group, "R1") | str_detect(Group, "-1"), "1", 
  ifelse(str_detect(Group, "R2") | str_detect(Group, "-2"), "2",
  ifelse(str_detect(Group, "R3"), "3", "NA"))))))) %>%
  
  #Add Pond info
  mutate(pond = ifelse(
  str_detect(Group,paste(jar_num$jar_no[c(1:3,10:12,19:21,28:30,37:39)], collapse = "|")), "1",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(4:6,13:15,22:24,31:33,40:42)], collapse = "|")), "2",
                     ifelse(
  str_detect(Group, paste(jar_num$jar_no[c(7:9,16:18,25:27,34:36,43:45)], collapse = "|")), "3", 
    ifelse(str_detect(Group, "P1"), "1", 
    ifelse(str_detect(Group, "P2"), "2",
    ifelse(str_detect(Group, "P3"), "3", "NA")))))))

Phycosphere_otu_D3 <- Phycosphere_otu_count %>% 
  filter((str_detect(Group,"-D3-") | str_detect(Group, "P")) & str_detect(Group,"022um"))


Phycosphere_otu_D31 <- Phycosphere_otu_count %>% 
  filter(str_detect(Group,"-D31-") & str_detect(Group,"022um"))

Phycosphere_otu_D3_3um <- Phycosphere_otu_count %>% 
  filter((str_detect(Group,"-D3-")) & str_detect(Group,"30um")) #D31 3.0um does not exist



```

```{r Phycosphere OTU D3/D31 Filtering 3 um}
#Only 3.0um data from D3

#Separate D3 and D31 data
agg_genus_data_D3_3um <- inner_join(Phycosphere_otu_D3_3um, Phycosphere_tax) %>% #Groups by otu automatically
  group_by(genus, Group, host_species, pond) %>% #Triplicate 1 is J1,2,3 and so on for each host species
  summarize(agg_rel_abund=sum(rel_abund)) %>% #Add together OTUs of the same group that are identical to get an aggregate abundance for each genus
  ungroup() %>%
  mutate(time = "D3") %>%
  mutate(genus=str_replace_all(string=genus,
    pattern="Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", replacement="A-N-P-R")) 


#D3 Algal Host Genus Makeup
colors <-   c(get_random_grid_colors(35, seed = 756810)) #Adjust the number to however many colors the graph says it needs upon error

top_genera_D3 <- agg_genus_data_D3_3um %>%
		group_by(genus) %>%
		summarize(median=median(agg_rel_abund)) %>%
		arrange((desc(median))) %>% # keep this so that the genera are sorted properly
		top_n(20, median) %>%
		pull(genus) # use pull to convert the names from a data frame to a vector of names
#view(top_genera_D3)

#D3 Algal Host Genus Makeup Stacked Barplot
plot_D3_3um <- agg_genus_data_D3_3um %>% filter (genus %in% top_genera_D3) %>%
  #filter(triplicate == 1 | triplicate == 4 | triplicate == 7 | triplicate == 10 | triplicate == 13) %>% #Triplicates from Pond one
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>% 
arrange(genus, desc(avg_rel_abund)) %>%
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))) %>%
#mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%
ggplot(aes(x = host_species, y = avg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),
    plot.title = element_text(size = 14))+
  guides(fill = guide_legend(size =10)) +
  labs( x = "Host Species", y = "Average Relative Abundance (Genus)", title = "Top 20 3.0um Day 3 Genera Abundance by Host Species", fill = "Genus")


#Combine D3 and D31 plots using ggpubr
ggarrange(plot_D3_3um + font("xlab", size = 10),
          common.legend = TRUE, legend = "bottom")
ggsave(filename = "D3-D31 Phycosphere Comparison 3.0um.png",  dpi = 300, path = "./figures", width = 10, height = 8)
```

```{r Phycosphere OTU D3/D31 Filtering 0.22 um}
#Separate D3 and D31 data
agg_genus_data_D3 <- inner_join(Phycosphere_otu_D3, Phycosphere_tax) %>% #Groups by otu automatically
  group_by(genus, Group, host_species, pond) %>% #Triplicate 1 is J1,2,3 and so on for each host species
  summarize(agg_rel_abund=sum(rel_abund)) %>% #Add together OTUs of the same group that are identical to get an aggregate abundance for each genus
  ungroup() %>%
  # group_by(genus, triplicate, host_species) %>% #Average triplicate counts for each genus
  # summarize(avg_rel_abund = mean(agg_rel_abund)) %>%
  # ungroup() %>%
  mutate(time = "D3") %>%
  mutate(genus=str_replace_all(string=genus,
    pattern="Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", replacement="A-N-P-R")) 

agg_genus_data_D31 <- inner_join(Phycosphere_otu_D31, Phycosphere_tax) %>% #Groups by otu automatically
  group_by(genus, Group, host_species, pond) %>% #Triplicate 1 is J1,2,3 and so on for each host species
  summarize(agg_rel_abund=sum(rel_abund)) %>% #Add together OTUs of the same group that are identical to get an aggregate abundance for each genus
  ungroup() %>%
  # group_by(genus, triplicate, host_species) %>% #Average triplicate counts for each genus
  # summarize(avg_rel_abund = mean(agg_rel_abund)) %>%
  # ungroup() %>%
  mutate(time = "D31") %>%
  mutate(genus=str_replace_all(string=genus,
    pattern="Allorhizobium-Neorhizobium-Pararhizobium-Rhizobium", replacement="A-N-P-R")) 


#D3 Algal Host Genus Makeup
colors <-   c(get_random_grid_colors(35, seed = 756810)) #Adjust the number to however many colors the graph says it needs upon error

# top_genera_D3_mean <- agg_genus_data_D3 %>%
# 		group_by(genus) %>%
# 		summarize(mean=mean(agg_rel_abund)) %>%
# 		arrange((desc(mean))) %>% # keep this so that the genera are sorted properly
# 		top_n(20, mean) %>%
# 		pull(genus) # use pull to convert the names from a data frame to a vector of names

top_genera_D3 <- agg_genus_data_D3 %>%
		group_by(genus) %>%
		summarize(median=median(agg_rel_abund)) %>%
		arrange((desc(median))) %>% # keep this so that the genera are sorted properly
		top_n(20, median) %>%
		pull(genus) # use pull to convert the names from a data frame to a vector of names
#view(top_genera_D3)


top_genera_D31 <- agg_genus_data_D31 %>%
		group_by(genus) %>%
		summarize(median=median(agg_rel_abund)) %>%
		arrange((desc(median))) %>% # keep this so that the genera are sorted properly
		top_n(20, median) %>%
		pull(genus) # use pull to convert the names from a data frame to a vector of names
#view(top_genera_D31)

```
```{r Phycosphere Comparison}
#D3 Algal Host Genus Makeup Stacked Barplot
plot_D3 <- agg_genus_data_D3 %>% filter (genus %in% top_genera_D3 | genus %in% top_genera_D31) %>%
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>% 
arrange(genus, desc(avg_rel_abund)) %>%
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))) %>%
#mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%
ggplot(aes(x = host_species, y = avg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),
    plot.title = element_text(size = 14))+
  guides(fill = guide_legend(size =10)) +
  labs( x = "Host Species", y = "Average Relative Abundance (Genus)", fill = "Genus")


#D31 Algal Host Genus Makeup Stacked Barplot
plot_D31 <- agg_genus_data_D31 %>% filter (genus %in% top_genera_D31 | genus %in% top_genera_D3) %>%
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
arrange(genus, desc(avg_rel_abund)) %>%
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))) %>%
#mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%
ggplot(aes(x = host_species, y = avg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),
    plot.title = element_text(size = 14))+
  guides(fill = guide_legend(size =10)) +
  labs( x = "Host Species", y = "Average Relative Abundance (Genus)", fill = "Genus")

#Combine D3 and D31 plots using ggpubr
ggarrange(plot_D3 + font("xlab", size = 10), plot_D31 + rremove("ylab") + font("xlab", size = 10),
          common.legend = TRUE, legend = "bottom", nrow = 1, ncol = 2, labels = "AUTO")
ggsave(filename = "D3-D31 Phycosphere Comparison.png",  dpi = 300, path = "./figures", width = 10, height = 8)

 #Recombine as one dataframe for contamination checks
agg_genus_data <- merge(agg_genus_data_D3, agg_genus_data_D31, all=TRUE) %>%
  merge(., agg_genus_data_D3_3um, all = TRUE)

  
# # unique genera
# length(unique(agg_genus_data$genus))

top_genera <- agg_genus_data %>%
		group_by(genus, time) %>%
		summarize(median=median(agg_rel_abund)) %>%
		arrange((desc(median))) %>% # keep this so that the genera are sorted properly
		top_n(20) %>%
		pull(genus) # use pull to convert the names from a data frame to a vector of names

agg_genus_data %>% filter (genus %in% top_genera_D3 | genus %in% top_genera_D31) %>%
  arrange(genus) %>%
  #mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%?
ggplot(aes(x = host_species, y = agg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 12))  +
  ylab("Relative Abundance (Genus) \n") +
  ggtitle("GRBC D31/D3 Isolates by Triplicate")

```

```{r GRBC Comparison}
#Filter genus hits from GRBC silva and freshwater database hits
Phycosphere_GRBC_D3 <- inner_join(GRBC_all_tax, agg_genus_data_D3) %>%
  filter(!str_detect(otu, "D31")) %>%
  group_by(genus, host_species, time, color) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
  select(genus, host_species, color)


Phycosphere_GRBC_D31 <- inner_join(GRBC_all_tax, agg_genus_data_D31) %>%
  filter(str_detect(otu, "D31"))  %>%
  group_by(genus, host_species, time, color) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
  select(genus, host_species, color)

#Isolates that are probably contaminants, not represented in phycosphere data
Phycosphere_GRBC_cont <- anti_join(GRBC_all_tax, agg_genus_data) %>%
  mutate(contaminated = TRUE)
Phycosphere_GRBC_match <- inner_join(GRBC_all_tax, agg_genus_data) %>%
  mutate(contaminated = FALSE)

Phycosphere_GRBC_cont %>% group_by(genus) %>%
  summarize(distinct_genus = n_distinct(genus))
write.xlsx( full_join(Phycosphere_GRBC_match, Phycosphere_GRBC_cont), "GRBC_Match&Cont.xlsx")

agg_genus_data_D3_color <- agg_genus_data_D3 %>% 
  #Filter out reads below a threshold (1/500 worked best) that are not part of our isolate collection
  filter(!(agg_rel_abund < 1/500) | genus %in% Phycosphere_GRBC_D3$genus) %>%
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
  left_join(., Phycosphere_GRBC_D3, by = c("genus", "host_species")) %>%
  mutate(color = if_else(condition = is.na(color), true = "black", false = color)) %>%
  mutate(time = "D3")  %>% 
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community")), genus = reorder_within(genus, -avg_rel_abund, host_species))

agg_genus_data_D31_color <- agg_genus_data_D31 %>% 
  #Filter out reads below the detection limit that are not part of our isolate collection
  filter(!(agg_rel_abund < 1/500) | genus %in% Phycosphere_GRBC_D31$genus) %>%
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
  left_join(., Phycosphere_GRBC_D3, by = c("genus", "host_species")) %>%
  mutate(color = if_else(condition = is.na(color), true = "black", false = color)) %>%
  mutate(time = "D3")  %>% 
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community")), genus = reorder_within(genus, -avg_rel_abund, host_species))




#Color schemes for each host species
color_chlorella <- agg_genus_data_D3_color %>%
  filter(host_species == "Chlorella") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_coelastrum <- agg_genus_data_D3_color %>%
  filter(host_species == "Coelastrum") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_scenedesmus <- agg_genus_data_D3_color %>%
  filter(host_species == "Scenedesmus") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_monoraphidium<- agg_genus_data_D3_color %>%
  filter(host_species == "Monoraphidium") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_selenastrum <- agg_genus_data_D3_color %>%
  filter(host_species == "Selenastrum") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))

chl_D3 <- agg_genus_data_D3_color %>% 
  filter(host_species == "Chlorella") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_chlorella$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = "none") +
  scale_fill_manual(values = c("#7FC97F")) +
  labs(y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()

coe_D3 <- agg_genus_data_D3_color %>% 
  filter(host_species == "Coelastrum") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, 
                               color = color_coelastrum$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = "none") +
  scale_fill_manual(values = c("#BEAED4")) +
  guides(fill = guide_legend(size = 10)) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.32), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()

sce_D3 <- agg_genus_data_D3_color %>% 
  filter(host_species == "Scenedesmus") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, 
                               color = color_scenedesmus$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = "none") +
  scale_fill_manual(values = c("#E0115F")) +
  guides(fill = guide_legend(size = 10)) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()

mon_D3 <- agg_genus_data_D3_color %>% 
  filter(host_species == "Monoraphidium") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, 
                               color = color_monoraphidium$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = "none") +
  scale_fill_manual(values = c("#FDC086")) +
  guides(fill = guide_legend(size = 10)) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()

sel_D3 <- agg_genus_data_D3_color %>% 
  filter(host_species == "Selenastrum") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_selenastrum$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position='none') +
  #facet_wrap(~host_species, scales = "free_y", dir = "v") +
  scale_fill_manual(values = c("#386CB0")) +
  labs( y = "Relative Abundance (Genus)", x = "Genus",  fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()

nat_comm_D0 <- agg_genus_data_D3 %>% 
  #Filter out reads below the detection limit that are not part of our isolate collection
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
  filter(!(avg_rel_abund < 10e-4) | genus %in% Phycosphere_GRBC_D3$genus) %>%
  mutate(time = "D3")  %>% 
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community")), genus = reorder_within(genus, -avg_rel_abund, host_species)) %>%
  filter(host_species == "Natural Community") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = "black"),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = 'bottom',
    legend.spacing.x = unit(0, 'cm')) +
  #facet_wrap(~host_species, scales = "free_y", dir = "v") +
  scale_fill_manual(values = c("#D4D5B0")) +
  labs( y = "Relative Abundance (Genus)", x = "Genus",  fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered()


#Color schemes for each host species
color_chlorella <- agg_genus_data_D31_color %>%
  filter(host_species == "Chlorella") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_coelastrum <- agg_genus_data_D31_color %>%
  filter(host_species == "Coelastrum") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_scenedesmus <- agg_genus_data_D31_color %>%
  filter(host_species == "Scenedesmus") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_monoraphidium<- agg_genus_data_D31_color %>%
  filter(host_species == "Monoraphidium") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))
color_selenastrum <- agg_genus_data_D31_color %>%
  filter(host_species == "Selenastrum") %>%
  select(color, genus, avg_rel_abund) %>%
  arrange(desc(avg_rel_abund))

chl_D31 <- agg_genus_data_D31_color %>% 
  filter(host_species == "Chlorella") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_chlorella$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position="right",
    legend.spacing.x = unit(0, 'cm')) +
  scale_fill_manual(values = c("#7FC97F")) +
  labs(y = "Relative Abundance (Genus)", x = "Genus", fill = "Host Species") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered() 

coe_D31 <- agg_genus_data_D31_color %>% 
  filter(host_species == "Coelastrum") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_coelastrum$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position="bottom",
    legend.spacing.x = unit(0, 'cm')) +
  scale_fill_manual(values = c("#BEAED4")) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered() 

sce_D31 <- agg_genus_data_D31_color %>% 
  filter(host_species == "Scenedesmus") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = color)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  scale_fill_manual(values = c("#E0115F", "black"), name =  '', labels = c("Cultured Isolate", "Not Isolated")) +
  guides(fill = guide_legend(
    title = quote(paste(italic("Scenedesmus acuminatus"))),
    title.position = "top",
    nrow = 2, byrow = TRUE
  )) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = '',
        title = "Rank Abundance Curve Based on 16S v4 Tag Sequencing of Phycosphere Communities") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.2), breaks = c(0.0, 0.1, 0.2)) +
  theme(axis.title.x = element_text(size = 20),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 15,
                               color = color_scenedesmus$color),
    panel.background = element_rect(fill = "transparent"), # bg of the panel
    plot.background = element_rect(fill = "transparent", color = NA), # bg of the plot
    panel.grid.major = element_blank(), # get rid of major grid
    panel.grid.minor = element_blank(), # get rid of minor grid
    legend.background = element_rect(fill = "transparent"), # get rid of legend bg
    legend.box.background = element_rect(fill = "transparent"),
    axis.title.y = element_text(size = 15),
    axis.text.y = element_text(size = 15),
    legend.position = "right",
    legend.box.just = "center",
    legend.direction = "vertical",
    legend.title = element_text(vjust = .5, size = 16),
    legend.text = element_text(vjust = .5, size = 15),
    plot.title = element_text(hjust = .5, size = 20)) +
   scale_x_reordered()

 # expression(atop('Abundant and Rare Genera Isolated from '* italic('Scenedesmus acuminatus'), ' Phycosphere After 31 Days of Exposure to a Natural Pond Community'))
ggsave(filename = "Phycosphere Isolate Subset.png",  dpi = 400, path = "./figures", bg = "transparent", width = 15, height = 6)

mon_D31 <- agg_genus_data_D31_color %>% 
  filter(host_species == "Monoraphidium") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_monoraphidium$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position="bottom",
    legend.spacing.x = unit(0, 'cm')) +
  scale_fill_manual(values = c("#FDC086")) +
  labs( y = "Relative Abundance (Genus)", x = "Genus", fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered() 

sel_D31 <- agg_genus_data_D31_color %>% 
  filter(host_species == "Selenastrum") %>%
  #Reorder genus by average relative abundance to get a smooth curve
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
  geom_bar(stat = "Identity") +
  theme_minimal() +
  theme(axis.title.x = element_blank(),
    axis.text.x = element_text(angle = 70, hjust = 1, size = 9, color = color_selenastrum$color),
    axis.title.y = element_text(size = 8),
    axis.text.y = element_text(size = 10),
    legend.position = 'bottom',
    legend.spacing.x = unit(0, 'cm')) +
  scale_fill_manual(values = c("#386CB0")) +
  labs( y = "Relative Abundance (Genus)", x = "Genus",  fill = "") +
  scale_y_continuous("Average Relative Abundance", limits = c(0, 0.3), breaks = c(0.0, 0.1, 0.2, 0.3)) +
   scale_x_reordered() 

#Create a combined figure using patchwork

patchwork <- ((chl_D3 | chl_D31 + rremove("ylab")) / (coe_D3 | coe_D31 + rremove("ylab")) / (sce_D3 | sce_D31 + rremove("ylab")) / (mon_D3 | mon_D31 + rremove("ylab")) / (sel_D3 | sel_D31+ rremove("ylab")) / nat_comm_D0) 

patchwork + 
  plot_annotation(
    tag_levels = "A",
) & theme(
  legend.spacing = unit(0, 'cm'),
  legend.box.just = 'left',
  legend.text = element_text(size = 14),
  legend.title = element_text(size = 16),
  axis.text.x = element_text(size = 10)
)
ggsave(filename = "Isolates D3-D31 Phycosphere Comparison.png",  dpi = 300, path = "./figures", width = 15, height = 18)

#Creating a representative plot for the poster.

ggsave(filename = "Phycosphere Isolate Subset.png",  dpi = 400, path = "./figures", bg = "transparent", width = 16, height = 6)
```



```{r dodge barplot}
#Filter genus hits from GRBC silva and freshwater database hits
Phycosphere_GRBC_D3 <- inner_join(GRBC_all_tax, agg_genus_data_D3) %>%
  filter(!str_detect(otu, "D31")) %>%
  group_by(genus, host_species, time, color) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup()
 

Phycosphere_GRBC_D31 <- inner_join(GRBC_all_tax, agg_genus_data_D31) %>%
  filter(str_detect(otu, "D31"))  %>%
  group_by(genus, host_species, time, color) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() 

#Dodge barplot for GRBC isolates with abundance (incomplete because of inadequate sequencing)
Phycosphere_GRBC_D3  %>%
  filter(avg_rel_abund != 0) %>%
mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum"))) %>%
  mutate(genus = fct_reorder(genus, avg_rel_abund, .fun = 'sum', .desc = TRUE)) %>%
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
geom_bar(stat = "identity", position = "dodge") +
  scale_fill_manual(values = host_colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9))+
  guides(fill = guide_legend(size = 10)) +
  labs( y = "Relative Abundance (Genus)", title = "D3 Genus Abundance by Host Species", fill = "Genus")

Phycosphere_GRBC_D31  %>%
   filter(avg_rel_abund != 0) %>%
mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum"))) %>%
  mutate(genus = fct_reorder(genus, avg_rel_abund, .fun = 'sum', .desc = TRUE)) %>%
  ggplot( aes(x = genus, y = avg_rel_abund, fill = host_species)) +
geom_bar(stat = "identity", position = "dodge") +
  scale_fill_manual(values = host_colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),)+
  guides(fill = guide_legend(size = 10)) +
  labs( y = "Relative Abundance (Genus)", title = "Day 31 Genus Abundance by Host Species", fill = "Genus")
```


```{r Phycosphere analysis top genera}
agg_genus_data_D3 %>%
		group_by(genus, host_species) %>%
		summarize(avg_rel_abund = mean(agg_rel_abund)) %>%
		arrange(desc(avg_rel_abund)) %>%
  ungroup()

agg_genus_data_D3 %>%
		group_by(pond,genus, host_species) %>%
  summarize(avg_rel_abund = median(agg_rel_abund)) %>%
		arrange(desc(avg_rel_abund)) %>%
	filter(genus %in% top_genera_D3) %>%
  mutate(genus=factor(genus, levels=top_genera_D3)) %>%
	mutate(avg_rel_abund = avg_rel_abund + 1/21000) %>%
	ggplot(aes(x=genus, y=avg_rel_abund, color = host_species)) +
		geom_hline(yintercept=1/16050, color="gray") +
		geom_boxplot() + 
  scale_color_manual(name=NULL,
			values=c("purple", "blue", "red", "green", "orange", "beige"),
			breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"),
			labels=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community")) +
		labs(title="Top 2% of Genera of D3 022um Host Algae",
			x=NULL,
			y="Relative abundance") +
		scale_y_log10(breaks=c(1e-4, 1e-3, 1e-2, 1e-1, 1), labels=c(1e-2, 1e-1, 1, 10, 100)) +
		theme_classic() +
  coord_flip()

agg_genus_data_D31 %>%
  group_by(pond, genus, host_species) %>%
  summarize(avg_rel_abund = median(agg_rel_abund)) %>%
		arrange(desc(avg_rel_abund)) %>%
	filter(genus %in% top_genera_D31) %>%
  mutate(genus=factor(genus, levels=top_genera_D31)) %>%
	mutate(avg_rel_abund = avg_rel_abund + 1/21000) %>%
	ggplot(aes(x=genus, y=avg_rel_abund, color = host_species)) +
		geom_hline(yintercept= 1/5570, color="gray", show.legent = TRUE) +
		geom_boxplot() +
 scale_color_manual(name=NULL,
			values=c("purple", "blue", "red", "green", "orange", "beige"),
			breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"),
			labels=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))  +
		labs(title="Top 2% of Genera of D31 022um Host Algae",
			x=NULL,
			y="Relative abundance") +
  scale_y_log10(breaks=c(1e-4, 1e-3, 1e-2, 1e-1, 1), labels=c(1e-2, 1e-1, 1, 10, 100)) +
		theme_classic() +
  coord_flip()
```
```{r stats}
genera_tests_D31 <- agg_genus_data_D31 %>%
					nest(sample_data = c(-genus)) %>%
					mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~host_species, data=.)))) %>%
					unnest(test) %>%
					mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
					arrange(p.value.adj)

sig_genera_D31 <- genera_tests_D31 %>%
					filter(p.value.adj <= 0.05) %>%
					pull(genus)

genera_tests_D3 <- agg_genus_data_D3 %>%
					nest(sample_data = c(-genus)) %>%
					mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~host_species, data=.)))) %>%
					unnest(test) %>%
					mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
					arrange(p.value.adj)

sig_genera_D3 <- genera_tests_D3 %>%
					filter(p.value.adj <= 0.05) %>%
					pull(genus)

genera_tests_D3_3um <- agg_genus_data_D3_3um %>%
					nest(sample_data = c(-genus)) %>%
					mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~host_species, data=.)))) %>%
					unnest(test) %>%
					mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
					arrange(p.value.adj)

sig_genera_D3_3um <- genera_tests_D3_3um %>%
					filter(p.value.adj <= 0.05) %>%
					pull(genus)

genera_tests_host <- agg_genus_data %>%
          nest(sample_data = c(-genus)) %>%
					mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~host_species, data=.)))) %>%
					unnest(test) %>%
					mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
					arrange(p.value.adj)

sig_genera_host <- genera_tests_host %>%
  filter(p.value.adj <= 0.05) %>%
					pull(genus)
# 
# genera_tests_time <- agg_genus_data %>%
#           nest(sample_data = c(-host_species)) %>%
# 					mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~time, data=.)))) %>%
# 					unnest(test) %>%
# 					mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
# 					arrange(p.value.adj)


#Plot significant data only

agg_genus_data_D3 %>%
	filter(genus %in% sig_genera_D3) %>%
	mutate(genus=factor(genus, levels=sig_genera_D3)) %>%
	mutate(avg_rel_abund=agg_rel_abund+1/21000) %>%
	ggplot(aes(x=genus, y=avg_rel_abund, color=host_species)) +
		geom_hline(yintercept=1/16050, color="gray") +
		geom_boxplot() +
		scale_color_manual(name=NULL,
			values=c("purple", "blue", "red", "green", "orange", "beige"),
			breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"),
			labels=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))  +
		labs(title="x genera are significantly associated with host species",
			x=NULL,
			y="Relative abundance") +
		scale_y_log10(breaks=c(1e-4, 1e-3, 1e-2, 1e-1, 1), labels=c(1e-2, 1e-1, 1, 10, 100)) +
		theme_classic() +
  coord_flip()


agg_genus_data_D3_3um %>%
	filter(genus %in% sig_genera_D3_3um) %>%
	mutate(genus=factor(genus, levels=sig_genera_D3_3um)) %>%
	mutate(avg_rel_abund=agg_rel_abund+1/21000) %>%
	ggplot(aes(x=genus, y=avg_rel_abund, color=host_species)) +
		geom_hline(yintercept=1/16050, color="gray") +
		geom_boxplot() +
		scale_color_manual(name=NULL,
			values=c("purple", "blue", "red", "green", "orange", "beige"),
			breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"),
			labels=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))  +
		labs(title="x genera are significantly associated with host species",
			x=NULL,
			y="Relative abundance") +
		scale_y_log10(breaks=c(1e-4, 1e-3, 1e-2, 1e-1, 1), labels=c(1e-2, 1e-1, 1, 10, 100)) +
		theme_classic() +
  coord_flip()

agg_genus_data_D31 %>%
	filter(genus %in% sig_genera_D31) %>%
	mutate(genus=factor(genus, levels=sig_genera_D31)) %>%
	mutate(avg_rel_abund=agg_rel_abund+1/21000) %>%
	ggplot(aes(x=genus, y=avg_rel_abund, color=host_species)) +
		geom_hline(yintercept=1/16050, color="gray") +
		geom_boxplot() +
		scale_color_manual(name=NULL,
			values=c("purple", "blue", "red", "green", "orange", "beige"),
			breaks=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"),
			labels=c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))  +
		labs(title="x genera are significantly associated with host species",
			x=NULL,
			y="Relative abundance") +
		scale_y_log10(breaks=c(1e-4, 1e-3, 1e-2, 1e-1, 1), labels=c(1e-2, 1e-1, 1, 10, 100)) +
		theme_classic() +
  coord_flip()

#D3 Algal Host Genus Makeup Stacked Barplot
plot_D3 <- agg_genus_data_D3 %>% filter (genus %in% sig_genera_D3) %>%
  #filter(triplicate == 1 | triplicate == 4 | triplicate == 7 | triplicate == 10 | triplicate == 13) %>% #Triplicates from Pond one
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>% 
arrange(genus, desc(avg_rel_abund)) %>%
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))) %>%
#mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%
ggplot(aes(x = host_species, y = avg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),
    plot.title = element_text(size = 14))+
  guides(fill = guide_legend(size =10)) +
  labs( x = "Host Species", y = "Average Relative Abundance (Genus)", title = "Top 20 Day 3 Genera Abundance by Host Species", fill = "Genus")


#D31 Algal Host Genus Makeup Stacked Barplot
plot_D31 <- agg_genus_data_D31 %>% filter (genus %in% top_genera_D31 | genus %in% top_genera_D3) %>%
  #filter(triplicate == 1 | triplicate == 4 | triplicate == 7 | triplicate == 10 | triplicate == 13) %>% #Triplicates from Pond one
  group_by(genus, host_species) %>%
  summarize(avg_rel_abund = mean(agg_rel_abund)) %>% #Take average of all the groups for one host species
  ungroup() %>%
arrange(genus, desc(avg_rel_abund)) %>%
  mutate(host_species = factor(host_species, levels = c("Chlorella", "Coelastrum", "Scenedesmus", "Monoraphidium", "Selenastrum", "Natural Community"))) %>%
#mutate(triplicate = fct_reorder(triplicate, as.numeric(triplicate))) %>%
ggplot(aes(x = host_species, y = avg_rel_abund, fill = genus)) + 
  geom_bar(stat = "identity", position = "stack") +
  scale_fill_manual(values = colors) +
  theme(axis.title.x = element_blank(), 
    axis.text.x = element_text(angle = 70, hjust = 1, size = 10),
    axis.text.y = element_text(size = 9),
    axis.title.y = element_text(size = 9),
    plot.title = element_text(size = 14))+
  guides(fill = guide_legend(size =10)) +
  labs( x = "Host Species", y = "Average Relative Abundance (Genus)", title = "Top 20 Day 31 Genera Abundance by Host Species", fill = "Genus")

#Combine D3 and D31 plots using ggpubr
ggarrange(plot_D3 + font("xlab", size = 10), plot_D31 + rremove("ylab") + font("xlab", size = 10), common.legend = TRUE, legend = "bottom", nrow = 1, ncol = 2, labels = "AUTO")
ggsave(filename = "D3-D31 Phycosphere Comparison2.png",  dpi = 300, path = "./figures", width = 10, height = 8)

```

```{r Subset}
#Remove numbers in parentheses from the table to aid in classification.

# Phycosphere_tax <- as.data.frame(lapply(Phycosphere_tax , gsub, pattern = "\\s*\\([^\\)]+\\)", replacement = ""))
# 
# GRBC_Freshwater2016_final <- as.data.frame(lapply(GRBC_Freshwater2016_pruned, gsub, pattern = "\\s*\\([^\\)]+\\)", replacement = ""))
# 
# GRBC_Silva_132_final <- as.data.frame(lapply(GRBC_Silva_132_pruned, gsub, pattern = "\\s*\\([^\\)]+\\)", replacement = ""))

```

